### PR TITLE
MWPW-140359 : Nala automation scripts for Merch card (catalog) variant

### PR DESCRIPTION
**Pull request by skumar09**
[MWPW-140359](https://jira.corp.adobe.com/browse/MWPW-140359 ): Merch card (Catalog) tests scripts

PR includes 
- Automation test scripts for the following features of the Merch card (Catalog) tests scripts
    - Merch-card (catalog)
    - Merch-card (catalog) with badge
    - Merch-card (catalog) with more info and badge

### CODE REVIEW

**Review by JackySun9 — APPROVED** [review]

File: tests/milo/merchcard.block.test.js [hunk 1]
```diff
@@ -300,7 +300,124 @@ test.describe('Milo Modal feature test suite', () => {
...
+      await page.waitForLoadState('domcontentloaded');
+      await expect(page).toHaveURL(`${baseURL}${features[8].path}`);    
+    });
+
+    await test.step('step-2: Verify Merch Card catalog content/specs', async () => {
+      await expect(await merchCard.catalog).toBeVisible();
+      await expect(await merchCard.catalogCardTitleH3).toContainText(data.titleH3);
+      await expect(await merchCard.catalogCardTitleH4).toContainText(data.titleH4);
+
+      //await expect(await merchCard.price).toContainText(data.price);
```
Comment (amauch-adobe) [inline tests/milo/merchcard.block.test.js:322]: Is this line needed for testing and such?  Just noticed it was commented out.

File: tests/milo/merchcard.block.test.js [hunk 2]
```diff
@@ -300,7 +300,124 @@ test.describe('Milo Modal feature test suite', () => {
...
+      await expect(await merchCard.catalog).toBeVisible();
+      
+      await expect(await merchCard.catalog).toHaveAttribute('badge-background-color', data.badgeBgColor);
+      await expect(await merchCard.catalog).toHaveAttribute('badge-color', data.badgeColor);;
+      await expect(await merchCard.catalog).toHaveAttribute('badge-text', data.badgeText);
+
+      await expect(await merchCard.catalogCardTitleH3).toContainText(data.titleH3);
+      await expect(await merchCard.catalogCardTitleH4).toContainText(data.titleH4);
+
+      //await expect(await merchCard.price).toContainText(data.price);
```
Comment (amauch-adobe) [inline tests/milo/merchcard.block.test.js:396]: same question here, it isn't a big deal, but just curious.

**Review by amauch-adobe — APPROVED** [review]
Approved.

File: tests/milo/merchcard.block.test.js (hunk 1 above)
Comment (skumar09) [reply tests/milo/merchcard.block.test.js:322]: @amauch-adobe  : It is commented because on PR branch prices won't display.

**Pull request merged**